tests/system/api/test_artifacts.py

```diff
@@ -91,3 +92,45 @@ def test_verify_artifact_tag_in_output(self):
         assert "v2" in outputs_uri, "Expected 'v2' tag in outputs_uri"
 
         mlrun.get_dataitem(output_uri)
+
+    def test_get_artifact_without_tag(self):
+        # Check the user flow of storing the same artifact twice without a tag and retrieving the artifact
+        key = "dummy-key"
+        self.project.log_artifact(key, body="123")
+        self.project.log_artifact(key, body="123")
+
+        artifacts = self.project.list_artifacts()
+        assert len(artifacts) == 2
+
+        non_latest_artifacts = [
+            artifact
+            for artifact in artifacts
+            if artifact["metadata"].get("tag") != "latest"
+        ]
+        assert (
+            len(non_latest_artifacts) == 1
+        ), "There should be exactly 1 non-latest artifact"
+
+        artifact = non_latest_artifacts[0]
+        key = artifact["metadata"]["key"]
+        tree = artifact["metadata"]["tree"]
+        iter = artifact["metadata"]["iter"]
+
+        retrieved_artifact = self.project.get_artifact(
```

Comment:
Few comments about the test:
1. you need to verify you got the old artifact and not the last one (possibly change the body between versions)
2. you need to have more than 2 artifacts in the DB. what happens if you have 3 artifacts? should you get 1? I believe the current code will explode in ` db_artifact = query.one_or_none()`
3. Why system test? this can be tested with unit test

Reply:
done